activerecord/lib/active_record/attributes.rb

```diff
@@ -280,7 +280,7 @@ def define_default_attribute(name, value, type, from_user:)
               name,
               value,
               type,
-              _default_attributes.fetch(name.to_s) { nil },
+              _default_attributes.fetch(name.to_s, nil),
```

Comment:
If `nil` is the default, you can just use `[]`.

```suggestion
              _default_attributes[name.to_s],
```

Reply:
@natematykiewicz Thank you, but I don't think we should change this yet unless there is a specific rule we can apply that supports it. I'd like to avoid making cosmetic changes without a "failing" case.

Reply:
I see you closed it because there's not a good way to differentiate `Rails.cache` from `Hash`. Makes sense. It's still possible this specific line using `[]` would be a performance change and not a cosmetic one. But you'd need a benchmark to see if it's actually faster.

Reply:
Yes, that is solving a different problem. Feel free to open a PR with benchmarks :pray: